### PR TITLE
NOTICK Update release-notes about database transaction limitations

### PR DESCRIPTION
**Pull request by lankydan**

### CODE REVIEW

**Review by rick-r3 — DISMISSED** [review]
Okay with me.  

The side effect might be relatively intrusive, but we're looking to make it better in 4.4.1

File: docs/source/release-notes.rst [hunk 1]
```diff
@@ -59,6 +59,10 @@ Given the addition of new APIs, the platform version of Corda 4.4 has been bumpe
 
 For more information on platform version, please see :doc:`versioning`. For more details on upgrading a CorDapp to use platform version 5, please see :doc:`app-upgrade-notes`.
 
+Known Issues
+~~~~~~~~~~~~
+
+Changes introduced in Corda 4.4 to increase reliability and correctness of a node's ledger, have highlighted limitations regarding database transactions. To prevent flows from continuing to process after a database transaction has failed to commit or suffered from a pre-commit persistence exception, extra database flushes have been added. These extra flushes can cause exceptions to be thrown where they were not before. Realistically, the overall behaviour between 4.3 and 4.4 will not have drastically changed but there will be scenarios where different exceptions are thrown or arise at points where they did not before. This limitation only effects flows that are catching database exceptions and then attempting to perform further database operations.
```
Comment (VCAMP) [inline docs/source/release-notes.rst]: @lankydan if I remember the issue correctly, we need to explicitly mention that users should _not_ expect to be able to catch database exceptions and perform further DB ops. I'd rephrase to:

"Changes introduced in Corda 4.4 to increase ledger integrity have highlighted limitations regarding database transactions. To prevent flows from continuing to process after a database transaction has failed to commit or suffered from a pre-commit persistence exception, extra database flushes have been added. These extra flushes can cause exceptions to be thrown where they were not before (or cause different exception types to be raised compared to Corda 4.3 or previous versions). In general, CorDapp developers should not expect to be able to catch exceptions thrown during a database transaction and then continue with further DB operations as part of the same flow. A safer pattern involves allowing the flow to fail and be retried".

Comment (rick-r3) [reply docs/source/release-notes.rst]: We want to try and relax this for `ConstraintViolationException` in 4.4.1 and 4.5.  Do we want to work that distinction into this in some way?  I know we don't like promising the future but we are hoping to "improve" what happens here to re-allow some patterns of use.  We're not 100% certain what we can and can't do though so it would have to be vague.  Is there some way we could ask folks to get in touch (with product, support) if they are adversely impacted so we can understand the use case?  They will do so "in the future" so we can then tell them about any enhancements in 4.4.1 or later releases then without having to promise here.

Comment (lankydan) [reply docs/source/release-notes.rst]: Do we have any decision on what we want to do here?

Comment (VCAMP) [reply docs/source/release-notes.rst]: @rick-r3 @lankydan  I'd still avoid mentioning it in the release notes (as you said, we don't make promises about the future in official docs). We can speak with the PS teams and with support so they're aware of the issue and can gather use cases if needed.

Comment (VCAMP) [reply docs/source/release-notes.rst]: @lankydan any thoughts on the suggested edit?

Comment (lankydan) [reply docs/source/release-notes.rst]: Personally, I am fine with the edits. I guess that means I need to go change it since it's my PR :)

**Review by VCAMP — APPROVED** [review]
